conditional/util/ldap.py

```diff
@@ -1,250 +1,120 @@
-    new_entry = new_entry.encode('utf-8')
-    new_results['member'].append(new_entry)
-    ldap_modlist = ldap.modlist.modifyModlist(old_results, new_results)
-    groupdn = "cn=%s,%s" % (group, group_search_ou)
-    ldap_conn.modify_s(groupdn, ldap_modlist)
-
-
-def _ldap_remove_member_from_group(username, group):
-    if read_only:
-        print("LDAP: Removing user %s from group %s" % (username, group))
-        return
-    if not _ldap_is_member_of_group(username, group):
-        return
-    ldap_results = ldap_conn.search_s(group_search_ou, ldap.SCOPE_SUBTREE,
-                                      "(cn=%s)" % group)
-    if len(ldap_results) != 1:
-        raise HousingLDAPError("Wrong number of results found for group %s." %
-                               group)
-    old_results = ldap_results[0][1]
-    new_results = copy.deepcopy(ldap_results[0][1])
-    new_results['member'] = [i for i in old_results['member'] if
-                             i.decode('utf-8') != "uid=%s,%s" % (username, user_search_ou)]
-    ldap_modlist = ldap.modlist.modifyModlist(old_results, new_results)
-    groupdn = "cn=%s,%s" % (group, group_search_ou)
-    ldap_conn.modify_s(groupdn, ldap_modlist)
-
-
-@ldap_init_required
-def _ldap_is_member_of_committee(username, committee):
-    ldap_results = ldap_conn.search_s(committee_search_ou, ldap.SCOPE_SUBTREE,
-                                      "(cn=%s)" % committee)
-    if len(ldap_results) != 1:
-        raise HousingLDAPError("Wrong number of results found for committee %s." %
-                               committee)
-    return "uid=" + username + "," + user_search_ou in \
-           [x.decode('ascii') for x in ldap_results[0][1]['head']]
+from conditional import ldap
 
 
-@lru_cache(maxsize=1024)
-def ldap_get_housing_points(username):
-    return int(_ldap_get_field(username, 'housingPoints'))
+def _ldap_get_group_members(group):
+    print("Got LDAP Group: {}".format(group))
+    return ldap.get_group(group).get_members()
+
+
+def _ldap_is_member_of_group(member, group):
+    group_list = member.get("memberOf")
+    for group_dn in group_list:
+        if group == group_dn.split(",")[0][3:]:
+            return True
+
 
+def _ldap_add_member_to_group(account, group):
+    if not _ldap_is_member_of_group(account, group):
+        ldap.get_group(group).add_member(account, dn=False)
 
-def ldap_get_room_number(username):
-    roomno = _ldap_get_field(username, 'roomNumber')
-    if roomno is None:
-        return "N/A"
-    return roomno.decode('utf-8')
+
+def _ldap_remove_member_from_group(account, group):
+    if _ldap_is_member_of_group(account, group):
+        ldap.get_group(group).del_member(account, dn=False)
 
 
 @lru_cache(maxsize=1024)
-def ldap_get_active_members():
-    return [x for x in ldap_get_current_students()
-            if ldap_is_active(x['uid'][0].decode('utf-8'))]
+def _ldap_is_member_of_committee(account, directorship):
```

Comment:
`committee` == `directorship`, although this isn't the only occurrence. Not sure if it's worth fixing.